### DISCUSSION TITLE
[Feature] Star rating for filtering, maybe color tagging, as in photo editors.

### DISCUSSION THREAD

**Discussion by Chreutz** [immich-app/immich#3619] [Feature Request]
### The feature

I have a lot of photos, some better than others.

Currently, if you want to have any prioritization of which images to show up, you need to make albums or just archive the ones you find uninteresting.

I would love if there was the option to give photos ratings, it's typically from 0 to 5, and then have a filter where you can select which ratings you want to see (preferably a sticky setting between visits). This could make the timeline much more clean, but also maintain the option to view everything. This is a common feature in photo editors such as darktable, digikam, lightroom.

Best regards

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by Chreutz** [comment]
I see some overlap with #3522 and #1651

**Comment by charles-997** [comment]
I think it would also be useful if Immich can read this 0-5 star rating from metadata (of course only applicable for images that already have this metadata from other software such as Lightroom, Digikam, etc)

**Reply by AngelaDMerkel** [reply]
I think this isn't a good approach because many Lightroom culling workflows and extensions use stars to help segregate/filter images in a way which doesn't reflect their quality. One extension I use in my workflow marks images as four stars, for example, and reading that in Immich as a reflection of quality probably isn't the best move.

**Reply by Chreutz** [reply]
Honestly, isn't that an issue with that extension? AFAIK, the star rating system in Lightroom (and others) is exactly for reflecting the quality?

**Reply by charles-997** [reply]
> I think this isn't a good approach because many Lightroom culling workflows and extensions use stars to help segregate/filter images in a way which doesn't reflect their quality. One extension I use in my workflow marks images as four stars, for example, and reading that in Immich as a reflection of quality probably isn't the best move. 

I think this is a good point that highlights the importance of user-choice.  If this feature is implemented, I think there will be use-cases where users wish to disable it.

(I also think disabled by default is reasonable and users can enable it as they see fit.)

**Comment by Chreutz** [comment]
Yes, that would be *really* nice, since you could use either for tagging/rating, as long as you're working in the same directories, and it would be in sync. It could bring immich into a semi-pro workflow.

**Comment by bo0tzz** [comment]
Added in #11580

**Comment by gschintgen** [comment]
This issue has been closed as implemented, but if I'm not mistaken the crucial part is still missing: actually making use of the rating, for example as a filter when searching. As far as I can tell (in version 1.123.0) I can filter a search by e.g. camera model or favorite status but not by rating. So maybe this feature request could be reopened?

**Reply by bo0tzz** [reply]
#11777
